charts/csp-rancher-usage-operator/crds/susecloud.net_cspadapterusagerecords.yaml

```diff
@@ -5,20 +5,23 @@ metadata:
   annotations:
     controller-gen.kubebuilder.io/version: v0.11.3
   creationTimestamp: null
-  name: productusages.usagerecords.suse.com
+  name: cspadapterusagerecords.susecloud.net
 spec:
-  group: usagerecords.suse.com
+  group: susecloud.net
   names:
-    kind: ProductUsage
-    listKind: ProductUsageList
-    plural: productusages
-    singular: productusage
+    kind: CSPAdapterUsageRecord
+    listKind: CSPAdapterUsageRecordList
+    plural: cspadapterusagerecords
+    singular: cspadapterusagerecord
+    shortNames:
+    - caur
   scope: Cluster
   versions:
   - name: v1
     schema:
       openAPIV3Schema:
-        description: ProductUsage is the Schema for the productusages API
+        description: CSPAdapterUsageRecord is the Schema for the cspadapterusagerecords
+          API
         properties:
```

Comment:
Shouldn't these properties be nested under `spec`? General fields are usually not kept at the top level.

Resources have a model with certain sections that include...
- metadata for metadata
- `spec` for all properties that are specified. This is for the things that are declared.
- `status` where controllers report status of where things are at.

For example, a `spec` section would be a good place to put the `base_product` and fields like that. And, a `status` section could record the last time this information was synced outside the cluster.

There are more details at https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/

Reply:
We are not really implementing a traditional Kubernetes controller where it is responsible for watching over a resource and then reconcileate. The adapter and usage operator merely waits up an an interval and update the resource based on managed node count. In that sense, the adapter and usage operator are not Kubernetes controllers.

Reply:
Also, per my understanding from @smarlowucf, the areas of impact for NV for introducing `spec` and `status` change will be rather significant?

Reply:
I brought it up from NV adapter perspective only because of time constraints. Following convention for CRD seems like a good idea but it was found too late. In this case it would require changes to internal code and external facing APIs. Given it doesn't seem to cause any trouble it's a pretty low priority task.